### DISCUSSION TITLE
For C# cartesian chart can we customize chart background for specific X axis ranges

### DISCUSSION THREAD

**Discussion by preintel** [Live-Charts/LiveCharts2#223] [Q&A]
I have code which produces the attached chart (top image), my code gets OHLC data from broker and creates the candlestick chart. The user can pull historical chart data (day/week/month every 1min/5min/etc) and I need to customize the chart for each one (mainly x axis needs to be customized). Since there is no data when market is closed there will be gaps between the candlesticks, is there a way to change the background of the chart in these areas to a solid color (see bottom image colored black areas), or is there a way to create a vertical line at specific points in x-axis to act a separator?
If there are any prebuilt themes for WinForms I can use for candlestick charts that would also be welcome.

Code creating the chart:
```
ViewModel viewM = new ViewModel();
                            viewM.Series = new ObservableCollection<ISeries>
                                                     {
                                                          new CandlesticksSeries<FinancialPoint>
                                                         {
                                                              Name = "AAPL",
                                                              Values = fPList,
                                                              TooltipLabelFormatter = p => "AAPL"+(new DateTime((long)p.SecondaryValue).ToString("hh:mm tt"))+$", H: {p.PrimaryValue:N2}, O: {p.TertiaryValue:N2}, C: {p.QuaternaryValue:N2}, L: {p.QuinaryValue:N2}"
                                                         }
                                                    };

                            viewM.XAxes = new[]
                                 {
                                                                    new Axis
                                                                    {
                                                                        Name = "Time",
                                                                        LabelsRotation = 15,
                                                                        Labeler = value => new DateTime((long)value).ToString("hh:mm tt"),
                                                                        UnitWidth = TimeSpan.FromMinutes(1).Ticks
                                                                    }
                                                                  ;

                            viewM.YAxes = new[]
                                      {
                                                                    new Axis
                                                                    {
                                                                        Name = "Price",
                                                                        Labeler = value => "$"+value.ToString(),
                                                                        MaxLimit = maxPrice + (0.01*maxPrice),
                                                                        MinLimit = minPrice - (0.01*minPrice)
                                                                    }
                                                                };
```

Example chart (3 days of 1 minute candlestick data for AAPL stock, spaces between days are when market is closed and no data):
<img width="676" alt="ExampleChart" src="https://user-images.githubusercontent.com/91435681/135732590-63aa9fbb-5a24-4b99-8bfe-2ff15428801f.png">

**Comment by CodeWithMichal** [comment]
You can do it using Sections.
https://github.com/beto-rodriguez/LiveCharts2/blob/master/samples/ViewModelsSamples/General/Sections/ViewModel.cs

Just define values for Xi and Xj and provide Fill property

**Reply by preintel** [reply]
Thanks I think this will do what I need to but my chart X axis is DateTime and section.cs only takes double values for Xi and Xj so not sure how I can create the sections on the chart. Do I need a mapper to map the X values or do I need a separate inherited class of Sections to handle the X values? 

I also can't pass DateTimePoint as a Type to RectangularSection since the class is nontypable, should I create a separate class to handle this or will this be added in a future update?

```
public ObservableCollection<RectangularSection> Sections { get; set; }
           = new ObservableCollection<RectangularSection>
           {
                new RectangularSection<DateTimePoint>      /* Error here using type DateTimePoint unless I create typable RectangularSection class */
                {
                    // creates a section
                    Xi = 1,   //Unable to pass DateTime here
                    Xj = 5,
                    Fill = new LiveChartsCore.SkiaSharpView.Painting.SolidColorPaint(new SkiaSharp.SKColor(255, 205, 210))
                },
           };
```

**Comment by CodeWithMichal** [comment]
In the end values on axis are translated to float or double. Here you can find default mappings: https://github.com/beto-rodriguez/LiveCharts2/blob/18adf1b555672289bcc89afb506236b43d44600e/src/LiveChartsCore/Kernel/LiveChartsSettings.cs

Just try to use Ticks from your datetime

**Reply by wasilus** [reply]
Hello,
It is possible to not display sections of DateTime  X axis? as in the pictures above.
I want to display only the sections of X axis which have data on Y axis.
